src/common/buffer.cc

```diff
@@ -997,9 +997,11 @@ static ceph::spinlock debug_lock;
         length -= round_length;
         other_buf += round_length;
       }
+      if (length == 0) {
+        return true;
+      }
```

Comment:
i don't think this helps. since we assume `length` should be identical fo the total length of the bufferlist. so returning early does not help with the performance, personally, neither does it help with readability.

Reply:
@tchaikov 
I agree with abandoning the change here.
BTW, since ```length``` should be identical to the total length of the bufferlist, do we still need to calculate ```const auto round_length = std::min<size_t>(length, bp.length());```? Could we use bp.length() as round_length directly.

Reply:
@rzarzynski i agree with @changchengx that there is no need to have
```c++
const auto round_length = std::min<size_t>(length, bp.length());
```
what about changing it to
```c++
assert(bp.length() <= length);
```

or just drop it?

Reply:
@rzarzynski If there's no other concern, I'll follow kefu's suggestion to change it.

Reply:
@changchengx, @tchaikov: yup, I think it's good idea to change just to an assertion.